File: works/tmp/1011.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class Main
{
    static long dfs(long distance, long moved, long cnt)
    {
        if(distance == 0) return cnt;

        for(long move = moved+1; move >= moved-1; move--)
        {
            if(move <= 0) continue;
            long sum = (move * (move + 1)) / 2;
            if(sum <= distance)
            {
                return dfs(distance-move, move, cnt+1);
            }
        }
        return 0;
    }

    public static void main(String[] args) throws IOException
    {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(br.readLine());
        String[] _s;
        while(t-- > 0)
        {
            _s = br.readLine().split(" ");
            long x = Integer.parseInt(_s[0]);
            long y = Integer.parseInt(_s[1]);
            long distance = y - x;
            long ans = dfs(distance, 0, 0);
            System.out.println(ans);
        }

    }

}

File: works/js/3-10/1011.js
let fs = require('fs');
let input = fs.readFileSync(0).toString().trim().split('\n');
// let input = fs.readFileSync('input.txt').toString().split('\n');

let answer = "" 
let tc = parseInt(input[0]) 
for(t = 1; t <= tc; t++) {
    var [x, y] = input[t].split(' ').map(Number) 
    var distance = y - x 
    var ans = dfs(distance, 0, 0) 
    answer += ans + '\n' 
}

console.log(answer) 


function dfs(distance, moved, cnt) {
    if(distance == 0) return cnt 

    for(move = moved+1; move >= moved-1; move--) {
        if(move <= 0) continue 
        let sum = Math.floor((move * (move+1)) / 2)
        if(sum <= distance) {
            return dfs(distance-move, move, cnt+1) 
        } 
    }
}
